pkg/endpoint/policy.go

```diff
@@ -172,21 +176,33 @@ func (e *Endpoint) regeneratePolicy(owner Owner) error {
 	// repository.
 	e.setNextPolicyRevision(revision)
 
-	logger := e.getLogger().WithFields(logrus.Fields{
-		"forcedRegeneration": forceRegeneration,
-	})
+	e.updatePolicyRegenerationStatistics(stats, forceRegeneration, retErr)
 
-	totalRegeneration, _ := safetime.TimeSinceSafe(regenerateStart, logger)
+	return nil
+}
 
-	logger.WithField(logfields.PolicyRegenerationTime, totalRegeneration.String()).
-		Debug("Completed endpoint policy recalculation")
+func (e *Endpoint) updatePolicyRegenerationStatistics(stats *policyRegenerationStatistics, forceRegeneration bool, err error) {
+	success := err == nil
 
-	regenerateTimeSec := totalRegeneration.Seconds()
-	metrics.PolicyRegenerationCount.Inc()
-	metrics.PolicyRegenerationTime.Add(regenerateTimeSec)
-	metrics.PolicyRegenerationTimeSquare.Add(math.Pow(regenerateTimeSec, 2))
+	stats.totalTime.End(success)
+	stats.success = success
 
-	return nil
+	stats.SendMetrics()
+
+	fields := logrus.Fields{
+		"waitingForIdentityCache":    stats.waitingForIdentityCache,
+		"waitingForPolicyRepository": stats.waitingForPolicyRepository,
+		"policyCalculation":          stats.policyCalculation,
+		"forcedRegeneration":         forceRegeneration,
+	}
+	scopedLog := e.getLogger().WithFields(fields)
+
+	if err != nil {
+		scopedLog.WithError(err).Warn("Regeneration of policy failed")
+		return
+	}
+
+	scopedLog.Info("Completed endpoint policy recalculation")
```

Comment:
Change this to Debug level.

Reply:
The `Completed endpoint regeneration` message is shown on the Info level (see line 415). I just wanted to be consistent.

Reply:
Oof, that should be debug as well - otherwise, for a cluster with a lot of churn, we will fill up the logs with such messages upon every regeneration event, which is not good. Can you change this log message to debug? I will change the other one in another PR.